omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/ConceptProposalResource2_0.java

```diff
@@ -0,0 +1,282 @@
+			description.addProperty("uuid");
+			description.addProperty("encounter", Representation.DEFAULT);
+			description.addProperty("obsConcept", Representation.DEFAULT);
+			description.addProperty("obs", Representation.DEFAULT);
+			description.addProperty("mappedConcept", Representation.DEFAULT);
+			description.addProperty("originalText");
+			description.addProperty("finalText");
+			description.addProperty("state");
+			description.addProperty("comments");
+			description.addProperty("creator", Representation.DEFAULT);
+			description.addProperty("dateCreated");
+			description.addProperty("changedBy", Representation.DEFAULT);
+			description.addProperty("dateChanged");
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof RefRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("encounter", Representation.REF);
+			description.addProperty("originalText");
+			description.addProperty("state");
+			description.addProperty("creator", Representation.REF);
+			description.addProperty("dateCreated");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("originalText");
+		description.addProperty("mappedConcept");
+		description.addProperty("encounter");
+		description.addProperty("obsConcept");
+		return description;
+	}
+
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("finalText");
+		description.addProperty("encounter");
+		description.addProperty("obsConcept");
+		description.addProperty("mappedConcept");
+		description.addProperty("comments");
+		return description;
+	}
+
+	@Override
+	public ConceptProposal getByUniqueId(String uniqueId) {
+		return Context.getConceptService().getConceptProposalByUuid(uniqueId);
+	}
+
+	@Override
+	protected PageableResult doGetAll(RequestContext context) throws ResponseException {
+		return new NeedsPaging<>(
+				Context.getConceptService().getAllConceptProposals(Boolean.parseBoolean(context.getParameter("includeCompleted"))),
+				context);
+	}
+
+	@Override
+	protected PageableResult doSearch(RequestContext context) {
+		boolean includeCompleted = Boolean.parseBoolean(context.getParameter("includeCompleted"));
+		String sortOrder = context.getParameter("sortOrder");
+		String sortOn = context.getParameter("sortOn");
+
+		List<ConceptProposal> allConceptProposals = Context.getConceptService().getAllConceptProposals(includeCompleted);
+
+		if (sortOn != null && sortOn.equals("originalText")) {
```

Comment:
Do we support only `originalText` for searching?

Reply:
LegacyUI also supports sorting by a number of occurrences. Should I implement it? Then, I would have to add a wrapper object that would contain a proposal and the number of occurrences. Also, the endpoint will have to return only unique proposals (with unique original names).

Reply:
Could there be any reason why you intentionally left it out?

Reply:
I thought that the REST API should not omit some objects in return. In this case, it will return only one proposal object per original text occurrence. Should I sort them by occurrences instead but still keep all the objects in the response? Or should I make a wrapper that will keep unique proposals with their occurrences count and return a list of these instead?

Reply:
How did the legacyui behave?

Reply:
LegacyUI returned only proposals with unique names. Its results also included a number of occurrences. I can mimic the same behavior here.

Reply:
Is that related `originalText`?

Reply:
Oh yeah, I meant originalText instead of name. So when you have two proposals with the same original name, the Legacy UI would just display one proposal with that name, which will have 2 occurrences.

Reply:
So, does this rest api implement all the functionality provided by the legacyui for concept proposal management?

Reply:
Not yet, I'm going to work on it now.

Reply:
Same pull request?

Reply:
Yes, I will update this PR

Reply:
@dkayiwa I've updated this PR by adding the ability to sort proposals by occurrences. This endpoint now also returns only proposals with the unique original text, as the Legacy UI does.

Reply:
@BartlomiejRasztabiga so are you now done with everything that the legacyui provided for concept proposal management?

Reply:
Yes